consensus/polybft/polybft.go

```
@@ -598,3 +599,38 @@ func (p *Polybft) GetBridgeProvider() consensus.BridgeDataProvider {
 func (p *Polybft) FilterExtra(extra []byte) ([]byte, error) {
 	return GetIbftExtraClean(extra)
 }
+
+func registerForksAndHandlers(forks *chain.Forks) error {
+	//nolint:godox
+	// TODO: update to real values and read from smart contract
+	availableForks := []forkmanager.ForkName{}
+	activeForks := []*forkmanager.ForkInfo{}
+	handlers := []forkmanager.ForkHandler{}
+
+	for name, block := range *forks {
+		if !chain.IsForkAvailable(name) {
+			panic(fmt.Errorf("fork is not available: %s", name)) //nolint:gocritic
```

Maybe it would be better to propagate the error and prevent node starting up if possible?

---

I have refactored this a little bit https://github.com/0xPolygon/polygon-edge/pull/1544/commits/40a2b88479dfdc32313392be35267ae3f850c15b